src/Rings/mpoly-graded.jl

```diff
@@ -1562,6 +1562,397 @@ end
 ### Homogenization and Dehomogenization
 ############################################################################
 
+## 2024-02-06  New UI to homogenization & dehomogenization
+## used partly the code from 2023-07 (see below)
+
+
+struct homogenizer
```

Comment:
This should probably be `Homogenizer`, i.e. camel case. Then `homogenizer` should be added later on as the user facing constructor.

Reply:
Good point!   Is there a particularly neat way to do this?  I presume the "internal" constructors must have names identical to that of the `struct`.

Reply:
Should be parametrized also?

Reply:
Please clarify about what should be parametrized and why?  I have not so much experience with parametrized `struct` in Julia.  Right the code seems to work OK, but maybe it would be better with a parameter?

Reply:
@JohnAAbbott : I suggest to meet on gather. Then we can look into the parametrizing business together.